x/oracle/keeper/keeper.go

```diff
@@ -62,6 +62,13 @@ type Keeper struct {
 	// indexes
 	idIndex *indexes.Multi[uint64, string, types.CurrencyPairState]
 
+	// numRemoves is the number of CPs removed in the previous block.
+	numRemoves collections.Item[uint64]
+
+	// removedCPsCounter is an in-memory counter for the number of CPs removed in the previous block.
+	// It is flushed to numRemovedCPs during the EndBlocker.
+	removedCPsCounter uint64
```

Comment:
Is there a reason why we don't just increment `numRemoves` instead of keeping a separate counter for this? PreBlock is executed before begin block so it might just be simpler to reset this in begin block and have it be accessible directly via the keeper when preblock is called in the next height.